source/slang/slang-emit-cpp.cpp

```diff
@@ -1779,6 +1794,23 @@ void CPPSourceEmitter::emitComInterface(IRInterfaceType* interfaceType)
     // Emit methods.
     m_writer->emit("\n{\n");
     m_writer->indent();
+    // Emit GUID.
+    m_writer->emit("SLANG_COM_INTERFACE(0x");
```

Comment:
Small point: I wonder if storing the guid as the native type rather than the string and then having functions that output/parse appropriately might slightly better. No ambiguity around validity, can compare quickly (no case issues etc), no additional allocation. Having functions to traffic between guids and strings is useful elsewhere.